.github/workflows/deploy-docs-to-gh-pages.yml

```diff
@@ -0,0 +1,53 @@
+name: deploy
+
+on:
+  pull_request:
```

Comment:
I'm not so sure about this workflow running on pull request. IMO it should be ran only once on pushes on the said branches.